src/orion/storage/track.py

```diff
@@ -0,0 +1,464 @@
+        return config
+
+    def update_experiment(self, experiment, where=None, **kwargs):
+        """Update a the fields of a given trials
+
+        Parameters
+        ----------
+        experiment: Experiment
+            Experiment object to update
+
+        where: Optional[dict]
+            constraint experiment must respect
+
+        **kwargs: dict
+            a dictionary of fields to update
+
+        Returns
+        -------
+        returns true if the underlying storage was updated
+
+        """
+        pass
+
+    def fetch_experiments(self, query):
+        """Fetch all experiments that match the query"""
+        return self.backend.fetch_groups(query)
+
+    def register_trial(self, trial):
+        """Create a new trial to be executed"""
+        stamp = datetime.datetime.utcnow()
+        trial.status = 'new'
+        trial.submit_time = stamp
+
+        metadata = dict()
+        metadata['params_types'] = {remove_leading_slash(p.name): p.type for p in trial. params}
+        metadata['submit_time'] = to_json(trial.submit_time)
+        metadata['end_time'] = to_json(trial.end_time)
+        metadata['worker'] = trial.worker
+        metadata['metric_types'] = {remove_leading_slash(p.name): p.type for p in trial.results}
+        metadata['metric_types'][self.objective] = 'objective'
+        metadata['heartbeat'] = stamp
+
+        metrics = defaultdict(list)
+        for p in trial.results:
+            metrics[p.name] = p.value
+
+        self.current_trial = self.backend.new_trial(TrackTrial(
+            _hash=trial.hash_name,
+            status=get_track_status('new'),
+            project_id=self.project.uid,
+            group_id=self.group.uid,
+            parameters={p.name: p.value for p in trial.params},
+            metadata=metadata,
+            metrics=metrics
+        ))
+        return TrialAdapter(self.current_trial, objective=self.objective)
+
+    def register_lie(self, trial):
+        """Register a *fake* trial created by the strategist.
+
+        The main difference between fake trial and original ones is the addition of a fake objective
+        result, and status being set to completed. The id of the fake trial is different than the id
+        of the original trial, but the original id can be computed using the hashcode on parameters
+        of the fake trial. See mod:`orion.core.worker.strategy` for more information and the
+        Strategist object and generation of fake trials.
+
+        Parameters
+        ----------
+        trial: `Trial` object
+            Fake trial to register in the database
+
+        """
+        pass
+
+    def fetch_trials(self, query, *args, **kwargs):
+        """Fetch all the trials that match the query"""
+        def sort_key(item):
+            submit_time = item.submit_time
+            if submit_time is None:
+                return 0
+            return submit_time
+
+        query = to_json(query)
+
+        new_query = {}
```

Comment:
The query conversion process should be extracted and tested separately.

Reply:
We should remove it entirely and move away for dict queries

Reply:
So what should we do now?

Reply:
Should merge & it will be refactored shortly

Reply:
Sounds risky as we don't know exactly when and how we will refactor. Meanwhile any change in the core that breaks this will go undetected.